lib/test/cdo/test_code_generation.rb

```diff
@@ -75,5 +75,14 @@ class CodeGenerationTest < Minitest::Test
       code = CodeGeneration.random_unique_code model: FakeModel, code_attribute: 'custom_code_attribute'
       assert_equal 'CCCC', code
     end
+
+    it 'checks deleted codes if model responds to :with_deleted' do
+      FakeModel.stubs(:with_deleted).returns(FakeModel)
+      FakeModel.with_deleted.expects(:exists?).with(has_key(:custom_code_attribute)).returns(true, true, false).times(3)
```

Comment:
Unfortunately it looks like this test will pass even without the code change, because `FakeModel.with_deleted` is stubbed to return `FakeModel`, so calls to `FakeModel.exists?` will satisfy the expectation on this line. You could fix this by having the stub return some other class `FakeModelWithDeleted` instead.

Reply:
😱 you're right, thank you for catching that! adding `FakeModelWithDeleted` works as expected